src/main/java/net/mcreator/element/types/Attribute.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2024, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.element.types;
+
+import net.mcreator.element.GeneratableElement;
+import net.mcreator.element.parts.EntityEntry;
+import net.mcreator.workspace.elements.ModElement;
+
+import java.util.List;
+
+public class Attribute extends GeneratableElement {
+
+	public String name;
+	public Double minValue;
+	public Double maxValue;
+	public Double defaultValue;
+	public Boolean persists;

```

Comment:
How is this handled in vanilla? Do any attribute base values persist on player death?

Do attribute modifiers also persist on death?

Reply:
attribute modifiers only persist if they're marked as permanent I think

Reply:
Turns out that on 1.21.1 all player attributes persist by default, while on 1.20.1 they don't. So I just removed the option and made them persist by default on 1.20.1 too.